### PR TITLE
Throw and embed everything sharp (shards, knives, swords)

### PR DESCRIPTION
**Pull request by K-Dynamic**
## About the PR

Adds embedding to a bunch of sharp objects where it would make sense, including:
- Glass shards, shivs and broken bottles
- Knives
- Scalpels
- Flatcaps
- Swords and katanas
- Hatchets and fireaxes

## Why / Balance

Added embedding property to items missing them (i.e. throwing dmg with no embed) for parity to match with spears and knives, meaning glass shards now embed on throwing. This sort of nerfs any glass cannons but also makes explosions with glass shards more interesting.

Added throwing damage to a lot of knives and swords because it seems to make sense, as well as adding some more depth to improvised combat. For instance, the ninja could throw their katana at an opponent then recall it. It also didn't make sense why only survival/combat knives could do throwing damage but not something like normal kitchen knives.

## Technical details

- `Resources/Prototypes/Entities/Objects/Materials/shards.yml` (glass shards now embed)
- `Resources/Prototypes/Entities/Objects/Misc/broken_bottle.yml` (broken bottle now embeds)
- `Resources/Prototypes/Entities/Objects/Specific/Medical/surgery.yml` (scalpels do throwing damage and embed)
- `Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml` (added throwing damage and embed to a bunch of shivs and kitchen knives)
- `Resources/Prototypes/Entities/Objects/Weapons/Melee/mining.yml` (added throwing damage and embed to crusher dagger)
- `Resources/Prototypes/Entities/Objects/Weapons/Melee/sword.yml` (added throwing damage and embed to all swords)

## Media

image

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

N/A

**Changelog**

:cl:
- add: items under the category of swords (including katanas, sabres and machetes), knives (including glass shivs, kitchen knives, kukris) now deal throwing damage and embed in mobs.
- add: axes, hatchets, scythes, crusher daggers, and scalpels now deal throwing damage and embed in mobs.
- tweak: glass shards and broken bottles now embed in mobs.

### CODE REVIEW

**Comment by NoElkaTheGod** [comment]
I think something like a kitchen knive shouldn't have throwing damage. Have you ever tried to throw a kitchen knife? That thing is not really balanced for throwing. That's why there are special knives balanced for throwing. The same applies to swords and axes, tomahawks (throwing axes) exist for a reason.

On the other hand, making more throwables embeddable makes a lot of sense. Why does a combat knife embed on throw, but a glass shard flies right through? Is it so sharp it makes a hole through your entire body? Pretty sure it doesn't.

TLDR: make more things embed good, make more things throwable bad.

**Comment by thebadman4662** [comment]
> Have you ever tried to throw a kitchen knife? That thing is not really balanced for throwing.

Have you ever tried to throw a *spessman* kitchen knife though? Those totally are made for throwing!

> That's why there are special knives balanced for throwing. 

And as a perk they IGNORE armor from what I recall, reusable deathsquad killing.

**Comment by NoElkaTheGod** [comment]
> > Have you ever tried to throw a kitchen knife? That thing is not really balanced for throwing.
> 
> Have you ever tried to throw a _spessman_ kitchen knife though? Those totally are made for throwing!

Perhaps, but then what's the point of having two different knives, if they have the same stats? (curently the only difference is that a kitchen knife doesn't have throwing damage)

> > That's why there are special knives balanced for throwing.
> 
> And as a perk they IGNORE armor from what I recall, reusable deathsquad killing.

Yup, they do. Makes me wonder if that's the case with all the throwing weapons, or throwing knives specificaly? If it is universal, then throwing a captain's saber will be pretty damn good.

**Comment by NoElkaTheGod** [comment]
> Also ss14 makes everything more dangerous :trollface:

Everything except guns. You ain't surviving more than one good shot from a pistol lmao.

**Comment by thebadman4662** [comment]
> > Also ss14 makes everything more dangerous :trollface:
> 
> Everything except guns. You ain't surviving more than one good shot from a pistol lmao.

Meanwhile all those people surviving 60 pistol rounds because they are so high on adrenaline because pistol neither had stopping power and combatant had bleed out to be incapacitated/dead.

Not to mention how inaccurate shotguns are in SS14, clearly they should be sniper rifles just like in real life up to 100-200 tiles.

**Comment by superjj18** [comment]
> I think something like a kitchen knive shouldn't have throwing damage. Have you ever tried to throw a kitchen knife? That thing is not really balanced for throwing. That's why there are special knives balanced for throwing. The same applies to swords and axes, tomahawks (throwing axes) exist for a reason.
> 
> 
> 
> On the other hand, making more throwables embeddable makes a lot of sense. Why does a combat knife embed on throw, but a glass shard flies right through? Is it so sharp it makes a hole through your entire body? Pretty sure it doesn't.
> 
> 
> 
> TLDR: make more things embed good, make more things throwable bad.

Have you ever had a kitchen knife thrown at you?

**Comment by NoElkaTheGod** [comment]
> Have you ever had a kitchen knife thrown at you?

Surprisingly, yes i did. It hurt, but didn't embed at all. Still have a scar though.

**Comment by metalgearsloth** [comment]
I think embedding everything is overkill considering how janky it looks.

**Comment by EthanQix** [comment]
I like the concept of every sharp object having some throwing damage, with weapons actually made for throwing having more.

Also while it wouldn't be realistic, it would be funny to see a clown with a collection of swords embedded into them.

**Comment by slarticodefast** [comment]
Hey, I brought this up for maintainer dicussion and we decided not to merge this for the following reasons:
- Having too many embeddable objects will make people spam them.
- It makes actual throwing weapons like spears and throwing knifes less unique.
- There is a current bug that deletes embedded objects if the entity they are embedded into is destroyed.
- You can embedd a lot of glass shards into yourself and walk into someone who is barefoot to instantly kill them.
- embedding some of these objects looks weird visually because players have a circular hitbox which does not fit the humanoid sprite very well.
grafik
grafik
grafik

As such I will be closing this PR. Thank you for your contribution.
